libtiff/tests/test_libtiff_ctypes.py

```diff
@@ -576,341 +576,200 @@ def test_copy(tmp_path):
-    assert tiff.GetField('SubfileType') == lt.FILETYPE_REDUCEDIMAGE
-
-    tiff.SetField('TileWidth', 256)
-    assert tiff.GetField('TileWidth') == 256
-
-    tiff.SetField('TileLength', 256)
-    assert tiff.GetField('TileLength') == 256
-
-    # Test float tags
-    tiff.SetField('XResolution', 88.0)
-    assert tiff.GetField('XResolution') == 88.0
-
-    tiff.SetField('YResolution', 88.0)
-    assert tiff.GetField('YResolution') == 88.0
-
-    tiff.SetField('XPosition', 88.0)
-    assert tiff.GetField('XPosition') == 88.0
-
-    tiff.SetField('YPosition', 88.0)
-    assert tiff.GetField('YPosition') == 88.0
-
-    # Test double tags
-    tiff.SetField('SMaxSampleValue', 255.0)
-    assert tiff.GetField('SMaxSampleValue') == 255.0
-
-    tiff.SetField('SMinSampleValue', 0.0)
-    assert tiff.GetField('SMinSampleValue') == 0.0
-
-    # Test string tags
-    test_string = b"test string"
-    tiff.SetField('Artist', test_string)
-    assert tiff.GetField('Artist') == test_string
-
-    tiff.SetField('DateTime', test_string)
-    assert tiff.GetField('DateTime') == test_string
-
-    tiff.SetField('HostComputer', test_string)
-    assert tiff.GetField('HostComputer') == test_string
-
-    tiff.SetField('ImageDescription', test_string)
-    assert tiff.GetField('ImageDescription') == test_string
-
-    tiff.SetField('Make', test_string)
-    assert tiff.GetField('Make') == test_string
-
-    tiff.SetField('Model', test_string)
-    assert tiff.GetField('Model') == test_string
-
-    tiff.SetField('Software', test_string)
-    assert tiff.GetField('Software') == test_string
-
-    # Test setting a tag with a different data type
-    tiff.SetField('ImageWidth', 128)
-    assert tiff.GetField('ImageWidth') == 128
-
-    tiff.SetField('BitsPerSample', 8)
-    assert tiff.GetField('BitsPerSample') == 8
-
-    # Test tags with count > 1
-    colormap_red = list(range(256))
-    colormap_green = list(range(256))
-    colormap_blue = list(range(256))
-    tiff.SetField('ColorMap', [colormap_red, colormap_green, colormap_blue])
-    p_colormap_red, p_colormap_green, p_colormap_blue = tiff.GetField('ColorMap')
+    tiff = lt.TIFF.open(tmp_path / f'libtiff_set_get_field_lowlevel_{type_name}.tiff', mode='w')
+    try:
+        data_holders = {
+            'uint16': ctypes.c_uint16(0),
+            'uint32': ctypes.c_uint32(0),
+            'float': ctypes.c_float(0.0),
+            'double': ctypes.c_double(0.0),
+            'string': ctypes.c_char_p(b''),
+        }
+        data_holders_defaulted = {
+            'uint16': ctypes.c_uint16(0),
+            'uint32': ctypes.c_uint32(0),
+            'float': ctypes.c_float(0.0),
+            'double': ctypes.c_double(0.0),
+            'string': ctypes.c_char_p(b''),
+        }
+
+        test_data = tag_test_data[type_name]
+        data = data_holders[type_name]
+        p_data = ctypes.byref(data)
+        data_defaulted = data_holders_defaulted[type_name]
```

Comment:
You might be able to move this all to a fixture that itself is parametrized so that here, the fixture called (for example) "tiff_data_case" could be a tuple of `tiff, test_data, data, data_defaulted`. Similarly for the other test below.